src/pages/SettingsPage.js

```diff
@@ -0,0 +1,163 @@
+import Avatar from '../components/Avatar';
+import ROUTES from '../ROUTES';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import {version} from '../../package.json';
+
+const propTypes = {
+    // Safe area insets required for mobile devices margins
+    insets: SafeAreaInsetPropTypes.isRequired,
+
+    /* Onyx Props */
+
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+    // Display name of the current user from their personal details
+        displayName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatarURL: PropTypes.string,
+    }),
+
+    // Information about the network
+    network: PropTypes.shape({
+    // Is the network currently offline or not
+        isOffline: PropTypes.bool,
+    }),
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+    network: null,
+};
+
+let currentlyViewedReportID;
+let currentURL;
+let personalDetails;
+let email;
+
+Onyx.connect({
+    key: ONYXKEYS.CURRENTLY_VIEWED_REPORTID,
+    callback: val => (currentlyViewedReportID = val),
+});
+Onyx.connect({
+    key: ONYXKEYS.CURRENT_URL,
+    callback: val => (currentURL = val),
+});
+Onyx.connect({
+    key: ONYXKEYS.MY_PERSONAL_DETAILS,
+    callback: val => (personalDetails = val),
+});
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: val => email = val ? val.email : null,
+});
+
+const SettingsPage = ({myPersonalDetails, network, insets}) => {
+    const indicatorStyles = [
+        styles.settingsStatusIndicator,
+        network && network.isOffline
+            ? styles.statusIndicatorOffline
+            : styles.statusIndicatorOnline,
+    ];
+
+    // On the very first sign in or after clearing storage these
+    // details will not be present on the first render so we'll just
+    // return nothing for now.
+    if (!myPersonalDetails || _.isEmpty(myPersonalDetails)) {
+        return null;
+    }
+
+    return (
+        currentURL === '/settings' && (
+        <View
+            pointerEvents="box-none"
+            style={[
+                getSafeAreaMargins(insets),
+                styles.settingsPageBackground,
+
+            ]}
+        >
+            <View
+                style={[
+                    styles.flexColumn,
+                    styles.settingsPageContainer,
+                ]}
+            >
+                <HeaderWithCloseButton
+                    onCloseButtonPress={() => redirect(ROUTES.getReportRoute(currentlyViewedReportID))}
+                    title="Settings"
+                />
+                <View style={[styles.flexColumn, {
+                    width: '100%',
+                    alignItems: 'center',
+                    justifyContent: 'space-between',
+                }]}
+                >
+                    <View
+                        onPress={() => {
+                            redirect(ROUTES.SETTINGS);
```

Comment:
This might be a mistake since we are already on the settings page and a `View` doesn't need an `onPress`.